fix(gestures): fix the swipe and scrolling issues on touch devices"

mohammadrafigh opened the pull request:
Fixes #10187, #10145.

ThomasBurleson commented:
@mohammadrafigh - your Pull Request is failing with [Travis testing](https://travis-ci.org/angular/material/jobs/207551932). Please fix before we consider reviewing and merging.

mohammadrafigh commented:
@ThomasBurleson oops, thanks forgot to remove that one.

mohammadrafigh commented:
@ThomasBurleson oh no it's better to keep the tests for pointercancel event but instead expect .not.toHaveBeenCalled instead. I added back and fixed the test.

ThomasBurleson commented:
@DevVersion - please review.

EladBezalel commented on src/core/services/gesture/gesture.js:
```diff
@@ -643,7 +643,9 @@ function attachToDocument( $mdGesture, $$MdGestureHandler ) {
     pointer.endTime = +Date.now();
 
-    runHandlers('end', ev);
+    if (ev.type !== 'pointercancel') {
```
Can you shortly explain why?

mohammadrafigh replied:
Regarding to issues I mentioned in commit message, In chromium/chrome engine and webview in android there is a problem with angular material implementation of gestures. The problem is when we have swipable tabs with scrollable content dragging page to scroll fires pointercancel and causes to swipe to the other tab. The problem also can occure where ever we use drag and swipe together. So I excluded pointercancel from events to be handled by handler functions. If you want I can provide codepen to test, but the discussion on #10145 describes the issue well.

theLufenk replied:
Refer to our discussion in #10145

Pointercancel can be fired before pointerup. And that might mean that pointerup will not be fired. This is not good because a lot of cleanup and rollback logic often resides in the end event handler.

A fix of it might be browser detection based IMHO

EladBezalel commented:
@ThomasBurleson @jelbourn overall this fix seems legit, do we have a gestures expert somewhere ?

mohammadrafigh commented:
it's funny! the PR marked for 1.1.4 but 1.1.4 released and the PR is still here.

EladBezalel commented:
@mohammadrafigh though it was marked for 1.1.4, it seems like it didn't got merged in cause of the branch conflicts and got missed by the release, please rebase and let's get it in.

ThomasBurleson commented:
@mohammadrafigh - please rebase and squash your commits.

EladBezalel commented:
@mohammadrafigh can you please rebase and squash your commits?

mohammadrafigh commented:
@ThomasBurleson @EladBezalel sure I will do it tonight, sorry for late reply I was busy

mohammadrafigh commented:
@ThomasBurleson @EladBezalel did it

rhclayto commented:
Any news when 1.1.5 will come out with this new fix? It's much needed.

sdpillay commented:
Any update on this? much needed for us too.

rhclayto commented:
All checks passed, needs:merge. Does this mean this is ready to go in?

axedre commented:
C'mon people, make this happen! ;)

ThomasBurleson commented:
@rhclayto - preparing to merge for master. Thank you.

sdpillay commented:
Desperately waiting on this

axedre commented:
Ditto.


--
Andrea

2017-06-19 21:55 GMT+02:00 sdpillay <notifications@github.com>:

> Desperately waiting on this
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/angular/material/pull/10455#issuecomment-309555043>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ADgAL74vRSbRRqVS2H_l2u0wJKpoIVwYks5sFtJIgaJpZM4MS3fW>
> .
>

sdpillay commented:
Any update when this will make it?
